Testing/itkGPUGenericMultiResolutionPyramidImageFilterTest.cxx

```diff
@@ -45,6 +45,29 @@
 // The filter takes an input image and produces an output image.
 // We compare the CPU and GPU output image write RMSE and speed.
 
+template <typename FilterType>
+void
+UpdateFilterNTimes(typename FilterType::Pointer filter, const unsigned int N, const bool computeOnlyForCurrentLevel)
```

Comment:
Really nice! Not just because your helper function reduces the amount of duplicate code, but also because it makes it very clear that the test does exactly the same thing for the GPU filter as for the CPU filter!

Minor comments (no show stopper):
 - Our (ITK based) naming conversion is to have a `T` as first letter of a template type parameter: https://github.com/InsightSoftwareConsortium/ITKSoftwareGuide/blob/51b0eb02d945ab7f5f24f16139cdf00fd0c783a9/SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex#L1083 So in this case it could be `TFilter`.
 - Personally I would rather pass a function argument _by reference_ than by (smart or raw) pointer. In this case, I would declare the filter parameter something like `TFilter& filter`. However, I haven't been able not convince ITK developers in a similar case: https://github.com/InsightSoftwareConsortium/ITK/pull/2102#discussion_r519021533 

No action needed! 👍